starlite/contrib/sqlalchemy/repository.py

```diff
@@ -0,0 +1,270 @@
+
+        with wrap_sqlalchemy_exception():
+            result = await self._execute()
+            instances = list(result.scalars())
+            for instance in instances:
+                self.session.expunge(instance)
+            return instances
+
+    async def update(self, data: ModelT) -> ModelT:
+        """Update instance with the attribute values present on `data`.
+
+        Args:
+            data: An instance that should have a value for `self.id_attribute` that exists in the
+                collection.
+
+        Returns:
+            The updated instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found with same identifier as `data`.
+        """
+        with wrap_sqlalchemy_exception():
+            id_ = self.get_id_attribute_value(data)
+            # this will raise for not found, and will put the item in the session
+            await self.get(id_)
+            # this will merge the inbound data to the instance we just put in the session
+            instance = await self._attach_to_session(data, strategy="merge")
+            await self.session.flush()
+            await self.session.refresh(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def upsert(self, data: ModelT) -> ModelT:
+        """Update or create instance.
+
+        Updates instance with the attribute values present on `data`, or creates a new instance if
+        one doesn't exist.
+
+        Args:
+            data: Instance to update existing, or be created. Identifier used to determine if an
+                existing instance exists is the value of an attribute on `data` named as value of
+                `self.id_attribute`.
+
+        Returns:
+            The updated or created instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found with same identifier as `data`.
+        """
+        with wrap_sqlalchemy_exception():
+            instance = await self._attach_to_session(data, strategy="merge")
+            await self.session.flush()
+            await self.session.refresh(instance)
+            self.session.expunge(instance)
+            return instance
+
+    def filter_collection_by_kwargs(self, **kwargs: Any) -> None:
+        """Filter the collection by kwargs.
+
+        Args:
+            **kwargs: key/value pairs such that objects remaining in the collection after filtering
+                have the property that their attribute named `key` has value equal to `value`.
+        """
+        with wrap_sqlalchemy_exception():
+            self._select.filter_by(**kwargs)
+
+    @classmethod
+    async def check_health(cls, session: AsyncSession) -> bool:
+        """Perform a health check on the database.
+
+        Args:
+            session: through which we runa check statement
+
+        Returns:
+            `True` if healthy.
+        """
+        return (  # type:ignore[no-any-return]  # pragma: no cover
```

Comment:
Why is this complaining in type checking?

Reply:
> Returning Any from function declared to return "bool"

I think with the textual select, `scalar_one()` is typed to return `Any`.